C#/day12.cs

```diff
@@ -0,0 +1,23 @@
+class Grade : Student
+{
+    private int score;
+
+    public Grade(string fname, string lname, int phone, int score) : base(fname, lname, phone)
+    {
+        this.score = score;
+    }
+
+    public char calculate()
+    {
+        if (score < 40)
+            return 'D';
+        else if (score >= 40 && score < 60)
```

Comment:
For the sake of discussion: since the score ranges are contingents without gaps between them, and given the code generated by the compiler will execute the tests sequentially, you only need to check the upper bound of each range.